flumine/markets/blotter.py

```diff
@@ -67,6 +67,11 @@ def process_closed_market(self, market_book) -> None:
 
     def process_cleared_orders(self, cleared_orders) -> list:
         # todo update order.cleared?
+        for cleared_order in cleared_orders.orders:
+            order_id = cleared_order.customer_order_ref.split("-")[1]
+            if self.has_order(order_id):
+                self._orders[order_id].cleared_order = cleared_order
```

Comment:
And then:

```python
self[order_id].cleared_...
```

Reply:
I can make those changes no problem.

For the failed tests though I don't see what I'm meant to do.  `process_cleared_orders()` in the blotter is being called with a `Market` object from `baseflumine.py`, but I thought it should be being called with a `ClearedOrders` object.  Have I misinterpreted that?

Reply:
OK, couple of issues.

To change to

`order_id = current_order.customer_order_ref[STRATEGY_NAME_HASH_LENGTH + 1 :]`

I tried importing `STRATEGY_NAME_HASH_LENGTH` in blotter.py with

`from ..strategy.strategy import STRATEGY_NAME_HASH_LENGTH`

But that failed because there's a circular reference from blotter.py -> strategy.py -> market.py -> blotter.py.  Can the location that `STRATEGY_NAME_HASH_LENGTH` is defined be changed?

Secondly I changed base_flumine.py at line 258 to:

```
if self.BACKTEST or self.client.paper_trade:
    # Mock a ClearedOrders object
    mock_cleared_orders = mock.Mock()
    mock_cleared_orders.orders = []
    self._process_cleared_orders(mock_cleared_orders)
```

But that causes the `test__process_close_market_closed_paper()` test in test_baseflumine.py to fail:

    FAIL: test__process_close_market_closed_paper (test_baseflumine.BaseFlumineTest)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "C:\Users\arch4\AppData\Local\Programs\Python\Python39\lib\unittest\mock.py", line 1337, in patched
        return func(*newargs, **newkeywargs)
      File "C:\Users\arch4\source\repos\flumine\tests\test_baseflumine.py", line 295, in 
    test__process_close_market_closed_paper
        mock__process_cleared_orders.assert_called_with(
      File "C:\Users\arch4\AppData\Local\Programs\Python\Python39\lib\unittest\mock.py", line 907, in assert_called_with
        raise AssertionError(_error_message()) from cause
    AssertionError: expected call not found.
    Expected: _process_cleared_orders(<MagicMock name='events.ClearedOrdersEvent()' id='2188823838784'>)
    Actual: _process_cleared_orders(<Mock id='2188824336416'>)

I've not really used Mocks before so not sure how, or whether, the test should be updated.

Reply:
Yes move that default if required.

To be consistent with live I would do this:

```python
if self.BACKTEST or self.client.paper_trade:
    cleared_orders = resources.ClearedOrders(clearedOrders=[], moreAvailable=False)
    cleared_orders.market_id = market_id
    self._process_cleared_orders(events.ClearedOrdersEvent(cleared_orders))
```

Test `test__process_close_market_closed` then needs to be fixed with this:

```python
mock_market.blotter.process_cleared_orders.return_value = []
```

Reply:
I've made those changes by the way and all the tests pass now